src/controllers/company.number.controller.ts

```diff
@@ -52,6 +58,45 @@ const route = async (req: Request, res: Response, next: NextFunction): Promise<v
       templateName: templatePaths.COMPANY_NUMBER,
     });
   }
+
+  const companyNumber: string = req.body.companyNumber;
+  try {
+    logger.info(`Retrieving company profile for company number ${companyNumber}`);
+    const token: string = req.chSession.accessToken() as string;
+
+    // TODO: Place this call somewhere more global so that it not called on check company as well
+    const company: PTFCompanyProfile = await getCompanyProfile(companyNumber, token);
+
+    const existing = req.chSession.data[keys.PTF_SESSION];
+    if (existing) {
+      await sessionService.updatePTFSessionValue(req.chSession,
+          keys.COMPANY_NUMBER_IN_CONTEXT, company.companyNumber);
+    } else  {
+      await sessionService.createPromiseToFileSession(req.chSession, company.companyNumber);
+    }
+
+    return res.redirect(pageURLs.PTF_CHECK_COMPANY);
+  } catch (e) {
+    logger.error(`Error fetching company profile for company number ${companyNumber}`, e);
+    if (e.status === 404) {
+      buildError(res, errorMessages.COMPANY_NUMBER_NOT_FOUND);
```

Comment:
Looks like the wrong error message key is being used here - it's the company that's not found and not the company number.

Reply:
This was done as part of ticket LFA-1136 but will update this.